pydarn/io/borealis/borealis_array.py

```diff
@@ -149,101 +184,43 @@ def arrays(self):
-        Reads Borealis bfiq file that has been structured into arrays.
-
-        Returns
-        -------
-        arrays: dict
-            The Borealis data in a dictionary of arrays, according to the
-            restructured array file format.
-        """
-        pydarn_log.info("Reading Borealis bfiq file: {}"
-                        "".format(self.filename))
-        attribute_types = \
-            borealis_formats.BorealisBfiq.array_single_element_types()
-        dataset_types = borealis_formats.BorealisBfiq.array_array_dtypes()
-        unshared_fields = borealis_formats.BorealisBfiq.unshared_fields + \
-            borealis_formats.BorealisBfiq.array_only_fields
-        self._read_borealis_arrays(attribute_types,
-                                   dataset_types, unshared_fields)
-        return self._arrays
+        pydarn_log.info("Reading Borealis {} {} file: {}"
+                        "".format(self.borealis_version,
+                                  self.borealis_filetype, self.filename))
 
-    def read_rawacf(self) -> dict:
-        """
-        Reads Borealis rawacf file that has been structured into arrays.
+        attribute_types = self.format_class.array_single_element_types()
+        dataset_types = self.format_class.array_array_dtypes()
+        unshared_fields = self.format_class.unshared_fields()
 
-        Returns
-        -------
-        arrays: dict
-            The Borealis data in a dictionary of arrays, according to the
-            restructured array file format.
-        """
-        pydarn_log.info(
-            "Reading Borealis rawacf file: {}".format(self.filename))
-        attribute_types = \
-            borealis_formats.BorealisRawacf.array_single_element_types()
-        dataset_types = borealis_formats.BorealisRawacf.array_array_dtypes()
-        unshared_fields = borealis_formats.BorealisRawacf.unshared_fields + \
-            borealis_formats.BorealisRawacf.array_only_fields
-        self._read_borealis_arrays(attribute_types, dataset_types,
-                                   unshared_fields)
-        return self._arrays
-
-    def read_antennas_iq(self) -> dict:
-        """
-        Reads Borealis antennas_iq file that has been structured into arrays.
-
-        Returns
-        -------
-        arrays: dict
-            The Borealis data in a dictionary of arrays, according to the
-            restructured array file format.
-        """
-        pydarn_log.info("Reading Borealis antennas_iq file: {}"
-                        "".format(self.filename))
-        attribute_types = \
-            borealis_formats.BorealisAntennasIq.array_single_element_types()
-        dataset_types = \
-            borealis_formats.BorealisAntennasIq.array_array_dtypes()
-        unshared_fields = \
-            borealis_formats.BorealisAntennasIq.unshared_fields + \
-            borealis_formats.BorealisAntennasIq.array_only_fields
         self._read_borealis_arrays(attribute_types, dataset_types,
```

Comment:
``` suggestion
        self._read_arrays(attribute_types, dataset_types,
```

Reply:
I prefer just protected for most things here. I don't want to limit users that want to write subclasses, for example, to not be able to access this directly without the name mangling. Although I don't use this function in any subclass, there are others that I do use in subclasses.